### PR TITLE
#222 Either.fromPromise

### LINKED ISSUE #222: `Either.fromPromise`
I found that monet.js does not support async functions very well. How about adding 
`Either.fromPromise`? I can handle that...

### PR DESCRIPTION
**Pull request by wookieb**
Fixes #222

### CODE REVIEW

File: src/monet.d.ts [hunk 1]
```diff
@@ -250,6 +250,7 @@ interface IEitherStatic extends IMonadStatic {
   of: IRightStatic;    // alias for unit
   pure: IRightStatic;  // alias for unit
   fromTry<V, E = Error>(fn: () => V) : Either<E,V>;
+  fromPromise<V, E = Error>(promise: Promise<V>) : Promise<Either<V, E>>;
```
[the remark this answers is not in the document]
Comment (wookieb) [reply src/monet.d.ts]: Done

**Comment by ulfryk** [comment]
Rebase please ;)

**Comment by ulfryk** [comment]
All LGTM - only rebase needed (against `develop` branch)

**Comment by wookieb** [comment]
Sorry for the mess but now it should be fine I guess

**Pull request merged**